### PR TITLE
Makefile: add support for overriding the TAILSCALE_TRACK and SPK_BUILD

### PR DESCRIPTION
**Pull request by maisem**
Also document/use a new format for SPK_BUILD which is derived from the
TAILSCALE_VERSION.

### CODE REVIEW

**Comment by bradfitz** [comment]
What's the plan here? I thought we were deleting both these files.

**Comment by maisem** [comment]
> What's the plan here? I thought we were deleting both these files.

Yeah they will go away as soon as we get the go stuff over the line. There are a few dependency issues that i didn't want to deal with right now (e.g. getting the right version of go and compiling all the archs from source). These changes allow us to at least start using the builder and then we can continue to remove the shell scripts

File: Makefile [hunk 1]
```diff
@@ -1,7 +1,10 @@
 TAILSCALE_VERSION ?= "1.20.1"
-TAILSCALE_TRACK = "stable"
-# This needs to be monotinically increasing regardless of the TAILSCALE_VERSION
-SPK_BUILD = 17
+TAILSCALE_TRACK ?= "stable"
+# SPK_BUILD is derived from the TAILSCALE_VERSION using the forumla
+#   (major-1)*1e7 + minor*1e4 + patch*1e1 + dsm
+# e.g. for 1.20.1 (dsm7) => 200017
+# Note: The DSM version is appended by the build step.
+SPK_BUILD ?= 20001
```
Comment (bradfitz) [inline Makefile:7]: how about a default value of "0" or "1" or something obviously invalid?

File: Makefile [hunk 2]
```diff
@@ -1,7 +1,10 @@
 TAILSCALE_VERSION ?= "1.20.1"
-TAILSCALE_TRACK = "stable"
-# This needs to be monotinically increasing regardless of the TAILSCALE_VERSION
-SPK_BUILD = 17
+TAILSCALE_TRACK ?= "stable"
+# SPK_BUILD is derived from the TAILSCALE_VERSION using the forumla
+#   (major-1)*1e7 + minor*1e4 + patch*1e1 + dsm
+# e.g. for 1.20.1 (dsm7) => 200017
+# Note: The DSM version is appended by the build step.
```
Comment (bradfitz) [inline Makefile:6]: per comment on other PR, let's call it SPK_BUILD_BASE or SPK_BUILD_PREFIX or something if it's the part without the 6 or 7.

it's very confusing to have something called FOO in different places with different forms of FOO.

File: build-package.sh [hunk 3]
```diff
@@ -11,14 +11,12 @@ PACKAGE_CENTER_VERSION=$6
 
 PRIVILEGE_FILE="src/privilege-dsm${DSM_VERSION}"
 LOGROTATE_FILE="src/logrotate-dsm${DSM_VERSION}"
+SPK_BUILD="${SPK_BUILD}${DSM_VERSION}"
```
Comment (bradfitz) [inline build-package.sh:14]: ah, yeah, here it is. This sort of stuff scares me.

**Pull request closed without merging**